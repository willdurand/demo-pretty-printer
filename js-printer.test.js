const { renderDocument } = require("./document");
const { makeAST } = require("./js-ast");
const { makeDocument } = require("./js-printer");

describe(__filename, () => {
  const render = (text, options) => {
    const doc = makeDocument(makeAST(text), { originalText: text });

    return renderDocument(doc, options);
  };

  it.each(["var", "let"])("formats a variable declaration with %s", (kind) => {
    const text = `${kind} someLongVar;`;

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats a variable assignement", () => {
    const text = "const value = 123456789;";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats a variable assignement with a long array", () => {
    const text = "const value = [1,2,3,4,5,6,7,8,9];";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats function declarations", () => {
    const text = "function func() {return;}";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats function declarations with arguments", () => {
    const text = "function func(a, b, ccc) {return;}";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats arrow functions", () => {
    const text = "const foo = () => {};";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });

  it("formats arrow functions with arguments", () => {
    const text = "const foo = (a, bb, ccc) => {};";

    expect(render(text, { width: 50 })).toMatchSnapshot();
    expect(render(text, { width: 10 })).toMatchSnapshot();
  });
});
