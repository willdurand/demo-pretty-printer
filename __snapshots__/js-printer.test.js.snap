// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable assignement 1`] = `"const value = 123456789;"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable assignement 2`] = `
"const value =
  123456789;"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable assignement with a long array 1`] = `"const value = [1, 2, 3, 4, 5, 6, 7, 8, 9];"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable assignement with a long array 2`] = `
"const value = [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9
];"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable declaration with let 1`] = `"let someLongVar;"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable declaration with let 2`] = `"let someLongVar;"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable declaration with var 1`] = `"var someLongVar;"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats a variable declaration with var 2`] = `"var someLongVar;"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats arrow functions 1`] = `"const foo = () => {};"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats arrow functions 2`] = `"const foo = () => {};"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats arrow functions with arguments 1`] = `"const foo = (a, bb, ccc) => {};"`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats arrow functions with arguments 2`] = `
"const foo = (
  a,
  bb,
  ccc
) => {};"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats function declarations 1`] = `
"function func() {
  return;
}"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats function declarations 2`] = `
"function func() {
  return;
}"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats function declarations with arguments 1`] = `
"function func(a, b, ccc) {
  return;
}"
`;

exports[`/Users/william/projects/demo-pretty-printer/js-printer.test.js formats function declarations with arguments 2`] = `
"function func(
  a,
  b,
  ccc
) {
  return;
}"
`;
